create.ps1

```diff
@@ -1,194 +1,254 @@
+                $Exception.ErrorMessage = $ErrorObject.ErrorDetails.Message
+                break
+            }
+            "System.Net.WebException" {
+                $Exception.ErrorMessage = [System.IO.StreamReader]::new(
+                    $ErrorObject.Exception.Response.GetResponseStream()).ReadToEnd()
+                break
+            }
+            default {
+                $Exception.ErrorMessage = $ErrorObject.Exception.Message
+            }
+        }
+
+        $Exception.VerboseErrorMessage = @(
+            "Error at Line [$($ErrorObject.InvocationInfo.ScriptLineNumber)]: $($ErrorObject.InvocationInfo.Line)."
+            "ErrorMessage: $($Exception.ErrorMessage) [$($ErrorObject.ErrorDetails.Message)]"
+        ) -Join ' '
+
+        Write-Output $Exception
     }
 }
-#EndRegion
+#endregion functions
 
-if (-not($dryRun -eq $true)) {
-    try {
-        Write-Verbose 'Getting accessToken'
-        $splatGetTokenParams = @{
-            TenantId     = $config.TenantId
-            ClientId     = $config.ClientId
-            ClientSecret = $config.ClientSecret
-            Scope        = $config.Scope
-        }
-        $accessToken = (Get-LisaAccessToken @splatGetTokenParams).access_token
-        $authorizationHeaders = New-Object "System.Collections.Generic.Dictionary[[String],[String]]"
-        $authorizationHeaders.Add("Authorization", "Bearer $accessToken")
-        $authorizationHeaders.Add("Content-Type", "application/json")
-        $authorizationHeaders.Add("Mwp-Api-Version", "1.0")
-
-        # Create user
-        Write-Verbose "Creating KPN Lisa account for '$($personObj.DisplayName)'"
-        $splatParams = @{
-            Uri     = "$($config.BaseUrl)/Users?filter=startswith(userprincipalname,'$($account.userPrincipalName)')"
-            Method  = 'get'
-            Headers = $authorizationHeaders
-        }
-        $userResponse = Invoke-RestMethod @splatParams
-        if ($userResponse.count -eq 0) {
-            $splatParams = @{
-                Uri     = "$($config.BaseUrl)/Users"
-                Method  = 'POST'
-                Body    = ($account | ConvertTo-Json)
-                Headers = $authorizationHeaders
-            }
-            $userResponse = Invoke-RestMethod @splatParams
-            $objectId = $($userResponse.objectId)
-            $auditMessage = "Account '$($personObj.DisplayName)' Created. Id: '$objectId"
-
-            #Set Default WorkSpaceProfile
-            $workSpaceProfileGuid = "500708ea-b69f-4f6c-83fc-dd5f382c308b" #WorkspaceProfile   "friendlyDisplayName": "Ontzorgd"
-
-            $splatParams = @{
-                Uri     = "$($config.BaseUrl)/Users/$objectId/WorkspaceProfiles"
-                Method  = 'PUT'
-                Headers = $authorizationHeaders
-                body    = ($workSpaceProfileGuid | ConvertTo-Json)
+
+#region Aliasses
+$Config = $ActionContext.Configuration
+$Account = $OutputContext.Data
+$AuditLogs = $OutputContext.AuditLogs
```

Comment:
I changed away from all alias variables, they do work but I can see how this confuses people.

You say you have doubts this will work, but assigning an object to a variable will always be a reference in PowerShell, that is how this works and you have to copy the object to prevent this behavior.